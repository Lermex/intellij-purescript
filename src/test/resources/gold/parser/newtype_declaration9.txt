Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('Foo')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('Foo')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(ObjectType)
            PsiElement({)('{')
            PSASTWrapperElement(Row)
              <empty list>
            PsiElement(})('}')
            PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('Foo')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(TypeConstructor)
            PsiElement(proper name)('Foo')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PsiElement(()('(')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(ObjectType)
                PsiElement({)('{')
                PSASTWrapperElement(Row)
                  <empty list>
                PsiElement(})('}')
          PsiElement())(')')