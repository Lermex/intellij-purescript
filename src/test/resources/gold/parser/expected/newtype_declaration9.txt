Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('A')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSNewTypeDeclarationImpl(NEW_TYPE_DECLARATION)
        PsiElement(PSTokenType.newtype)('newtype')
        PSDeclarationTypeImpl(DECLARATION_TYPE)
          PsiElement(PSTokenType.PROPER_NAME)('Foo')
        PsiElement(PSTokenType.=)('=')
        PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
          PSTypeListImpl(TYPE_LIST)
            PSDeclarationTypeImpl(DECLARATION_TYPE)
              PsiElement(PSTokenType.PROPER_NAME)('Foo')
          PSDataDeclarationBodyImpl(DATA_DECLARATION_BODY)
            PsiElement(PSTokenType.{)('{')
            PsiElement(PSTokenType.})('}')
    PSDeclarationsImpl(DECLARATIONS)
      PSDeclarationImpl(DECLARATION)
        PSNewTypeDeclarationImpl(NEW_TYPE_DECLARATION)
          PsiElement(PSTokenType.newtype)('newtype')
          PSDeclarationTypeImpl(DECLARATION_TYPE)
            PsiElement(PSTokenType.PROPER_NAME)('Foo')
          PsiElement(PSTokenType.=)('=')
          PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
            PSTypeListImpl(TYPE_LIST)
              PSDeclarationTypeImpl(DECLARATION_TYPE)
                PsiElement(PSTokenType.PROPER_NAME)('Foo')
            PsiElement(PSTokenType.()('(')
            PSDataDeclarationBodyImpl(DATA_DECLARATION_BODY)
              PsiElement(PSTokenType.{)('{')
              PsiElement(PSTokenType.})('}')
            PsiElement(PSTokenType.))(')')